tests/userController.test.js

```diff
@@ -100,4 +100,71 @@ describe("user controller", () => {
       });
     });
   });
-});
+
+  describe("updates a user", () => {
+      it("updates an existent user", async() => {
+        const req = {
+          body: { id: 1, email: "newemail@example.com", password: "newpassword", permission: "newpermission" }
+        };
+        const res = {
+          status: jest.fn().mockReturnThis(),
+          send: jest.fn()
+        };
+    
+        database.User.update = jest.fn().mockResolvedValue([1]);
+    
+        await updateUser(req, res);
+    
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledWith("User data updated");
+      });
+      it("handles a request to update non existent user", async() => {
+          const req = {
+            body: { id: 99, email: "newemail@example.com", password: "newpassword", permission: "newpermission" }
+          };
+          const res = {
+            status: jest.fn().mockReturnThis(),
+            send: jest.fn()
+          };
+          
+          // O primeiro elemento da função updated do sequeleize é o número de linhas afetadas por essa atualização
+          // Se o valor de n em mockResolverValue([n]) for diferente de 0, não estaremos buscando por alterações
+          database.User.update = jest.fn().mockResolvedValue([0]);
+      
+          await updateUser(req, res);
+      
+          expect(res.status).toHaveBeenCalledWith(404);
+          expect(res.send).toHaveBeenCalledWith("User not found");
+      });
+      it("handles partial updates", async() => {
+        const req = {
+          body: { id: 69, email: "newemail@example.com" }
+        };
+        const res = {
+          status: jest.fn().mockReturnThis(),
+          send: jest.fn()
+        };
+
+        database.User.update = jest.fn().mockResolvedValue([69]);
+
+        await updateUser(req, res);
+
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledWith("User data updated");
+      });
+      it("handles database error", async() => {
+        const req = {
+          body: { id: 1, email: "newemail@example.com" }
+        };
+        const res = {
+          status: jest.fn().mockReturnThis(),
+          send: jest.fn()
+        };
+        database.User.update = jest.fn().mockResolvedValue(new Error("Database error"));
```

Comment:
Same as yesterday. No need to pass error  string to  the mocked error if you are already defining the error on the controller. You would define an error here if you were using `error.message` on the controller.

Reply:
I have removed the error string